app/frontend/src/features/profile/actions/RespondToFriendRequestProfileButton.tsx

```diff
@@ -0,0 +1,116 @@
+import { CircularProgress } from "@material-ui/core";
+import Button from "components/Button";
+import IconButton from "components/IconButton";
+import { CheckIcon, CloseIcon, PersonAddIcon } from "components/Icons";
+import Menu, { MenuItem } from "components/Menu";
+import { PENDING } from "features/connections/constants";
+import { SetMutationError } from "features/connections/friends";
+import useRespondToFriendRequest from "features/connections/friends/useRespondToFriendRequest";
+import { FriendRequest } from "pb/api_pb";
+import React, { useRef, useState } from "react";
+
+interface RespondToFriendRequestProfileButtonProps {
+  friendRequestId: number;
+  state: FriendRequest.FriendRequestStatus;
+  setMutationError: SetMutationError;
+}
+
+export const RESPOND_TO_FRIEND_REQUEST_MENU_ID =
+  "respond-to-friend-request-actions-menu";
+
+function RespondToFriendRequestProfileButton({
+  friendRequestId,
+  state,
+  setMutationError,
+}: RespondToFriendRequestProfileButtonProps) {
+  const [isOpen, setIsOpen] = useState({
+    accepted: false,
+    rejected: false,
+    menu: false,
+  });
+
+  const {
+    isLoading,
+    isSuccess,
+    reset,
+    respondToFriendRequest,
+  } = useRespondToFriendRequest();
+
+  const menuAnchor = useRef<HTMLButtonElement>(null);
+
+  const handleClick = (item: keyof typeof isOpen) => () => {
+    //close the menu if a menu item was selected
+    if (item !== "menu") {
+      setIsOpen((prevState) => ({ ...prevState, menu: false }));
+    } else {
+      setIsOpen((prevState) => ({ ...prevState, menu: true }));
+    }
+  };
+
+  const handleClose = (item: keyof typeof isOpen) => () => {
+    setIsOpen((prevState) => ({ ...prevState, [item]: false }));
+  };
+
+  return state === FriendRequest.FriendRequestStatus.PENDING ? (
+    <>
+      <Button
+        startIcon={<PersonAddIcon />}
+        innerRef={menuAnchor}
+        onClick={handleClick("menu")}
+        aria-controls={RESPOND_TO_FRIEND_REQUEST_MENU_ID}
+      >
+        {PENDING}
+      </Button>
+      <Menu
+        anchorEl={menuAnchor.current}
+        id={RESPOND_TO_FRIEND_REQUEST_MENU_ID}
+        onClose={handleClose("menu")}
+        open={isOpen.menu}
+      >
+        {" "}
+        {isLoading || isSuccess ? (
+          <CircularProgress />
+        ) : (
+          <>
+            <MenuItem onClick={handleClick("accepted")}>
+              {
+                <IconButton
+                  aria-label="Accept request"
+                  onClick={() => {
+                    reset();
```

Comment:
What does this reset do?

Reply:
I see it's copied from the other page. Try without - `reset` is used to clear the error and data, but since we're using `setMutationError` I don't think it's needed. If it works, remove it from the other page too :)

Reply:
I usually fake an error in chrome dev tools by going network->throttling->offline, then clicking